api/src/main/java/org/apache/iceberg/actions/RewriteManifests.java

```diff
@@ -44,6 +45,16 @@ public interface RewriteManifests
    */
   RewriteManifests rewriteIf(Predicate<ManifestFile> predicate);
 
+  /**
+   * Rewrite manifests in a given order, based on partition columns
+   *
+   * <p>If not set, manifests will be rewritten in the order of the table's spec.
+   *
+   * @param partitionSortOrder a list of partition field names
+   * @return this for method chaining
+   */
+  RewriteManifests sort(List<String> partitionSortOrder);
```

Comment:
yes that should have a default implementation to avoid adding a breaking change. See also https://github.com/apache/iceberg/blob/main/CONTRIBUTING.md#adding-a-default-implementation